### DISCUSSION TITLE
Bun Glob scan have different order than using node

### DISCUSSION THREAD

**Discussion by huyhoang160593** [oven-sh/bun#10112] [Q&A] [answered]
I have an fastify app which using autoload plugin to load all my plugins before create router. Which I have order for the plugins to load cause of dependency.
This is my original order that run fine in nodejs + tsx
```
00_sensible.plugin.ts
01_cors.plugin.ts
02_jwt.plugin.ts
03_drizzle.plugin.ts
04_bcrypt.plugin.ts
05_di.plugin.ts
```

But when I use bun, this is the order that bun load and they messes up my application
```
03_drizzle.plugin.ts
01_cors.plugin.ts
05_di.plugin.ts
00_sensible.plugin.ts
02_jwt.plugin.ts
04_bcrypt.plugin.ts
```

How can I solve this for running my application

**Comment by truongan07** [comment, marked as answer]
- I think Bun's Glob will asynchronous scan to get high performance task and return a promise. You can sort files in an array
```js
let list = [];

for await (const path of globalThis.Bun.Glob('./*/*').scan({ absolute: true })) list.push(path);
list = list.sort((find, string) => string.toString().trim().toLocaleLowerCase().localeCompare(find.toString().trim().toLocaleLowerCase())).reverse();

console.log(list);
```

**Comment by Jarred-Sumner** [comment]
Relying on filesystem order will likely cause issues like this. The operating system returns files in the order it wants, we have little control over that without manually sorting ourselves. On a different computer, it would return potentially in a different order. Your best bet is to sort them as @truongan07 shows
